src/event.c

```diff
@@ -148,13 +162,26 @@ void io_add_event(io_t *io, io_cb_t cb, void *data, WSAEVENT event) {
 }
 #endif
 
-void io_set(io_t *io, int flags) {
-#ifdef HAVE_SYS_EPOLL_H
+#ifdef HAVE_KQUEUE
+static void event_kqueue_init(void) {
+	if(!kq) {
+		kq = kqueue();
```

Comment:
We should have `event_*_exit()` functions as well that correctly clean up, otherwise static analysis or tools like Valgrind might complain.